docs/src/pages/guides/interoperability/interoperability.md

```diff
@@ -257,87 +298,88 @@ export default function StyledComponents() {
 
-The following example overrides the `thumb` style of `Slider` in addition to the custom styles on the slider itself.
+The following examples override the slider's `thumb` style in addition to the custom styles on the slider itself.
 
-{{"demo": "pages/guides/interoperability/StyledComponentsDeep.js"}}
+{{"demo": "pages/guides/interoperability/StyledComponentsDeep.js", "defaultCodeOpen": false}}
 
 ```jsx
 import * as React from 'react';
 import Slider from '@material-ui/lab/SliderStyled';
 import { experimentalStyled as styled } from '@material-ui/core/styles';
 
-const CustomizedSliderDeep = styled(Slider)`
+const CustomizedSlider = styled(Slider)`
   color: #20b2aa;
+
   :hover {
     color: #2e8b57;
   }
+
   & .MuiSlider-thumb {
     border-radius: 1px;
   }
 `;
 
-export default function CustomizedSliderDeepDemo() {
+export default function StyledComponentsDeep1() {
   return (
     <div>
       <Slider defaultValue={30} />
-      <CustomizedSliderDeep defaultValue={30} />
+      <CustomizedSlider defaultValue={30} />
     </div>
   );
 }
 ```
 
-The above demo relies on the [default `className` values](/styles/advanced/#with-material-ui-core), but you can provide your own class name: `.thumb`.
+The above demo relies on the [default `className` values](/styles/advanced/#with-material-ui-core), but you can provide your own class name with the `componentsProps` API.
 
 ```jsx
 import * as React from 'react';
 import { experimentalStyled as styled } from '@material-ui/core/styles';
 import Slider from '@material-ui/lab/SliderStyled';
 
-const StyledSlider = styled((props) => (
+const CustomizedSlider = styled((props) => (
   <Slider componentsProps={{ thumb: { className: 'thumb' } }} {...props} />
 ))`
   color: #20b2aa;
+
   :hover {
     color: #2e8b57;
   }
+
   & .thumb {
     border-radius: 1px;
   }
 `;
 
-export default function StyledComponentsDeep() {
+export default function StyledComponentsDeep2() {
   return (
     <div>
-      <Slider defaultValue={30}>Default</Slider>
-      <StyledSlider defaultValue={30}>Customized</StyledSlider>
+      <Slider defaultValue={30} />
+      <CustomizedSlider defaultValue={30} />
     </div>
   );
 }
 ```
 
 ### Theme
 
-Material-UI has a rich theme structure that you can take advantage of for
-color manipulations, transitions, media queries, and more.
-
 By using the Material-UI theme provider, the theme will be available in the theme context
 of the styled engine too (emotion or styled-components, depending on your configuration).
 
-You are encouraged to share the same theme object between Material-UI and your styles.
+> ⚠️ If you are **already** using a custom theme with styled-components or emotion,
+> it might not be compatible with Material-UI's theme specification. If it's not
+> compatible, you need to render the Material-UI's ThemeProvider <b>first</b>. This will
```

Comment:
Pick one:
```suggestion
> compatible, you need to render Material-UI's ThemeProvider <b>first</b>. This will
```
```suggestion
> compatible, you need to render Material-UI's ThemeProvider <b>first</b>. This will
```